tests/net-client-cache.rs

```diff
@@ -0,0 +1,344 @@
+#![cfg(feature = "net")]
+mod net;
+
+use crate::net::deckard::client::do_client;
+use crate::net::deckard::client::CurrStepValue;
+use crate::net::deckard::connect::Connect;
+use crate::net::deckard::parse_deckard::parse_file;
+use domain::base::{Dname, MessageBuilder, Rtype::Aaaa};
+use domain::net::client::cache;
+use domain::net::client::clock::{Clock, FakeClock};
+use domain::net::client::multi_stream;
+use domain::net::client::redundant;
+use domain::net::client::request::{
+    Error::NoTransportAvailable, RequestMessage, SendRequest,
+};
+use std::fs::File;
+use std::sync::Arc;
+
+const TEST_FILE_AD: &str = "test-data/client-cache/cache_ad.rpl";
```

Comment:
One can avoid the various `const FILE_NAME: &str = "file_path"` and N test fns that all invoke `async_test_cache()` by using a test parameterizer, e.g. like is done [here](https://github.com/NLnetLabs/domain/blob/5303cb215e3dabe6c4243de2b7dc672a43fdd260/tests/net-server.rs#L48).